src/dune_pkg/mount.ml

```diff
@@ -23,6 +23,29 @@ let of_opam_url loc url =
       >>| User_error.ok_exn
     in
     Git rev
+  | `Tar ->
+    let dir = Path.Build.(L.relative root [ "_fetch"; "url" ]) |> Path.build in
+    Source.fetch_archive_cached (loc, url)
+    >>= (function
+     | Error message_opt ->
+       let message =
+         match message_opt with
```

Comment:
This looks like a good place for `Option.map`.

Reply:
You meant `Option.value` no?

Reply:
Ah yes, indeed!